### PR TITLE
Bearlyml23, robo23, basic sync

### PR DESCRIPTION
**Pull request by jasonkimsoso**
robo23 and bearlyml23 setup

### CODE REVIEW

File: bearlyml23/bearlyml23.ld [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+OUTPUT_ARCH("riscv")
+
+/* Entry point */
+ENTRY(_start)
+
+MEMORY {
+  LIBIFROM (rwx): ORIGIN = 0x00020000, LENGTH = 64K
+  LIBIFRAM (rwx): ORIGIN = 0x00030000, LENGTH = 64K
+  SCRATCH  (rwx): ORIGIN = 0x08000000, LENGTH = 16K
```
Comment (T-K-233) [inline bearlyml23/bearlyml23.ld:52]: Scratchpad should be 64K?

File: bearlyml23/inc/bearlyml23.h [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+
+
+/* ================================ Peripheral Definitions ================================ */
+
+/* Peripheral Struct Definition */
+typedef struct {
+  __IO uint32_t MSIP0;                          /** MSIP Registers (1 bit wide) */
+  __IO uint32_t MSIP1;                          /** MSIP Registers (1 bit wide) */
+  __IO uint32_t MSIP2;                          /** MSIP Registers (1 bit wide) */
+  __IO uint32_t MSIP3;                          /** MSIP Registers (1 bit wide) */
```
Comment (T-K-233) [inline bearlyml23/inc/bearlyml23.h:94]: I helped you fix this. Bearlyml23 has only 4 cores, so there's only MSIP[0..3] registers

File: bearlyml23/inc/bearlyml23.h [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+} PLIC_TypeDef;
+
+// because the maximum struct size is 65535, we need to split PLIC content
+typedef struct {
+  PLIC_ContextControl_TypeDef context_controls[1024];
+} PLIC_Extra_TypeDef;
+
+
+typedef struct {
+  __IO uint32_t TILE0_RESET;      /** RocketTile 0 */   // 0x00
```
Comment (T-K-233) [inline bearlyml23/inc/bearlyml23.h:120]: Make sure the RCC (Reset and Clock Control) are correct. Do we still have reset and clock gate for all cores? How many cores are there?

File: robo23/inc/robo23.h [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+} PLIC_TypeDef;
+
+// because the maximum struct size is 65535, we need to split PLIC content
+typedef struct {
+  PLIC_ContextControl_TypeDef context_controls[1024];
+} PLIC_Extra_TypeDef;
+
+
+typedef struct {
+  __IO uint32_t TILE0_RESET;      /** RocketTile 0 */   // 0x00
```
Comment (T-K-233) [inline robo23/inc/robo23.h:119]: Same here as bearlyml23

**Review by T-K-233 — CHANGES_REQUESTED** [review]
Thanks!
Generally looks good. There's some small bugs that need to be fixed.

**Comment by jasonkimsoso** [comment]
Sorry about that. I didn't know PR automatically update the new commits.

**Review by T-K-233 — APPROVED** [review]
Looks good.

**Pull request merged**